src/pages/ImportKey.vue

```diff
@@ -35,51 +46,45 @@ q-page(padding)
 
 <script lang="ts">
 import { defineComponent } from "vue"
-import { Loading } from "quasar"
+import { LocalStorage } from "quasar"
 import { globalState as global } from "src/lib/global"
+import { invoke } from '@tauri-apps/api/tauri'
 const lang = global.data.loc.text.importKey
-import ms from "ms"
+
 export default defineComponent({
   data() {
     return {
-      keyInput: "",
+      rewardAddress: "",
       global: global.data,
       lang
     }
   },
-  computed: {
-    KeyStatusMsg(): string {
-      if (this.keyInput.length > 0) return lang.validKey
-      else return ""
-    },
-    validKey(): boolean {
-      if (this.keyInput.length > 0) return true
-      else return false
-    },
-    statusMsgStyle(): string[] {
-      if (this.validKey) return ["greenMsg"]
-      else return ["redMsg"]
-    }
-  },
   methods: {
-    importKey() {
-      Loading.show({
-        message: lang.importing,
-        boxClass: "bg-grey-2 text-grey-9"
-      })
-      setTimeout(() => {
-        Loading.show({
-          message: lang.imported,
-          boxClass: "bg-grey-2 text-grey-9",
-          spinnerColor: "green"
-        })
-      }, ms("2s"))
-      setTimeout(() => {
+    async validKey(): Promise<boolean> {
+      return invoke("check_reward_address_validity", { s: this.rewardAddress })
+    },
+    async importKey() {
+      if (await this.validKey) {
```

Comment:
Missing parentheses here:
```suggestion
      if (await this.validKey()) {
```

Reply:
Thanks Leo. But I am going to change this `validity check` function completely anyway, so that it won't require backend.